device.py

```diff
@@ -1,8 +1,38 @@
 from clip import Clip
 
 
-class Device():
+class DeviceOutput:
+    def __init__(self, method, name=None):
+        self.method = method
+        self.name = name or method.__name__
+        self.__doc__ = method.__doc__
 
+    def get_mapping(self, inst):
+        return inst.mapping
+
+    def __get__(self, inst, cls=None):
+        mapping = self.get_mapping(inst)
+        return mapping.setdefault(self.name, self.method(inst))
+
+    def __set__(self, inst, value):
+        mapping = self.get_mapping(inst)
+        mapping[self.name] = value
+        # inst.update_lookup()
+
+    def __delete__(self, inst):
```

Comment:
In learn dialog, we may add reset or undo button that will need to delete some value in temporary mapping. But, you're right it is not needed for now.